src/XamlX.IL.Cecil/CecilField.cs

```diff
@@ -6,41 +6,49 @@ namespace XamlX.TypeSystem
 {
     partial class CecilTypeSystem
     {
-        public class CecilField : IXamlField
+        class CecilField : IXamlField
         {
             private readonly FieldDefinition _def;
-            public CecilTypeSystem TypeSystem { get; }
-            public FieldReference Field { get; }
+            private readonly CecilTypeResolver _typeResolver;
+            public FieldDefinition Field { get; }
 
-            public CecilField(CecilTypeSystem typeSystem, FieldDefinition def, TypeReference declaringType)
+            public CecilField(CecilTypeResolver typeResolver, FieldDefinition def)
             {
-                TypeSystem = typeSystem;
+                _typeResolver = typeResolver;
                 _def = def;
-                Field = new FieldReference(def.Name, def.FieldType, declaringType);
```

Comment:
Is there any reason we have creating a copy of the FieldReference here from the definition?
Note, new type resolved doesn't mutate input references.

Reply:
Apparently, we needed FieldReference here, as passing FieldDefinition to the IL Emitter is a bad idea.